app/src/main/cpp/skyline/gpu.cpp

```diff
@@ -332,30 +333,27 @@ namespace skyline::gpu {
     }
 
     static PFN_vkGetInstanceProcAddr LoadVulkanDriver(const DeviceState &state) {
-        // Try turnip first, if not then fallback to regular with file redirect then plain dlopen
-        auto libvulkanHandle{adrenotools_open_libvulkan(
-            RTLD_NOW,
-            ADRENOTOOLS_DRIVER_CUSTOM,
-            nullptr, // We require Android 10 so don't need to supply
-            state.os->nativeLibraryPath.c_str(),
-            (state.os->privateAppFilesPath + "gpu/turnip/").c_str(),
-            "libvulkan_freedreno.so",
-            nullptr
-        )};
-        if (!libvulkanHandle) {
+        void *libvulkanHandle{};
+
+        // If the user has selected a custom driver, try to load it
+        if (!(*state.settings->gpuDriver).empty()) {
             libvulkanHandle = adrenotools_open_libvulkan(
                 RTLD_NOW,
-                ADRENOTOOLS_DRIVER_FILE_REDIRECT,
```

Comment:
you should do  ADRENOTOOLS_DRIVER_FILE_REDIRECT | ADRENOTOOLS_DRIVER_CUSTOM and file redirect into the public data dir

Reply:
> file redirect into the public data dir
Could you elaborate a bit?

Reply:
uh just do what was there before, pass in dir in the last param and pass in flags as `ADRENOTOOLS_DRIVER_FILE_REDIRECT | ADRENOTOOLS_DRIVER_CUSTOM`